libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -29,10 +29,26 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
     LOG(log_dg_) << "Received PBFT vote " << vote_hash;
 
     const auto vote_round = vote->getRound();
-    const auto pbft_round = pbft_mgr_->getPbftRound();
-    if (vote_round < pbft_round) {
+    const auto current_pbft_round = pbft_mgr_->getPbftRound();
+
+    // Check reward vote
+    if (vote_round < current_pbft_round && vote->getType() == cert_vote_type &&
+        vote->getBlockHash() == pbft_mgr_->getLastPbftBlockHash()) {
+      // Synchronization point in case multiple threads are processing the same vote at the same time
+      if (!seen_votes_.insert(vote_hash)) {
+        LOG(log_dg_) << "Received reward vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
+                     << ") already seen.";
+      } else if (vote_mgr_->addRewardVote(vote)) {
+        // As peers have small caches of known votes. Only mark gossiping votes
+        peer->markVoteAsKnown(vote_hash);
+        votes.push_back(std::move(vote));
+      }
+      continue;
+    }
```

Comment:
this should be inside of `if (vote_round < current_pbft_round) {` IMHO, as you 2x checking for same thing

Reply:
Fixed